docs/attributes-registry/gen-ai.md

```diff
@@ -0,0 +1,58 @@
+<!--- Hugo front matter used to generate the website version of this page:
+--->
+
+# Large Language Model (LLM)
+
+<!-- toc -->
+
+- [Generic LLM Attributes](#generic-llm-attributes)
+  - [Request Attributes](#request-attributes)
+  - [Response Attributes](#response-attributes)
+  - [Event Attributes](#event-attributes)
+
+<!-- tocstop -->
+
+## Generic LLM Attributes
+
+### Request Attributes
+
+<!-- semconv registry.gen_ai(omit_requirement_level,tag=llm-generic-request) -->
+| Attribute  | Type | Description  | Examples  |
+|---|---|---|---|
+| `gen_ai.request.max_tokens` | int | The maximum number of tokens the LLM generates for a request. | `100` |
+| `gen_ai.request.model` | string | The name of the LLM a request is being made to. | `gpt-4` |
+| `gen_ai.request.temperature` | double | The temperature setting for the LLM request. | `0.0` |
+| `gen_ai.request.top_p` | double | The top_p sampling setting for the LLM request. | `1.0` |
+| `gen_ai.system` | string | The name of the LLM foundation model vendor. | `openai` |
+
+`gen_ai.system` has the following list of well-known values. If one of them applies, then the respective value MUST be used, otherwise a custom value MAY be used.
+
+| Value  | Description |
+|---|---|
+| `openai` | OpenAI |
```

Comment:
Great point!

A few thoughts:
- I feel hesitant adding more systems before we modeled how the attributes would apply to them and if we need to add custom ones - created this issue https://github.com/open-telemetry/semantic-conventions/issues/839 so we don't forget to include it.
- I would avoid `azure-openai` - you can use vanilla OpenAI SDK to talk to Azure or Azure OpenAI (preview) to talk to vanilla openai and we'll never be able to guarantee the right value.
- Another problem with `google` - it's not specific enough and we don't have a good pattern there  for GCP services - https://github.com/open-telemetry/semantic-conventions/issues/608 

So I wonder if we can expand this list in a follow up PR or you consider it's really important to add right away?

Reply:
> Similar like previous, but while the SDK may be reusable across, the vendor/provider of the model could still be different. In a use case where langchain would be used, I still expect to see the actual vendor that is being used for a specific llm response, while the SDK may be langchain. The vendor value could be used for determining issues with a specific vendor or failover to other vendors in case of issues.

is there a reliable way to tell which vendor is behind the endpoint?

> Google is the vendor, while the exact product name/API's should be captured in another attribute?
I don't think it is critical to expand the list now, although I feel that adding at least 1 other could help keep the spec more agnostic.

in other semantic conventions we capture the name of the specific service like `gcp_pubsub` or `spanner`. Here we'd need to capture name name of google AI system. I'd love google folks to come up with the name.

Reply:
> is there a reliable way for an SDK to tell which vendor is behind the endpoint?

I understand your concern. In langchain it is seen as a "[provider](https://python.langchain.com/docs/integrations/platforms/)", and each provider is initialized by [importing a specific package](https://python.langchain.com/docs/integrations/platforms/openai), similar [approach for Llamaindex](https://docs.llamaindex.ai/en/stable/getting_started/starter_example_local/). Do you think these could be leveraged?

> in other semantic conventions we capture the name of the specific service like gcp_pubsub or spanner. Here we'd need to capture the name of google AI system. I'd love google folks to come up with the name. 
Agree, we may be able to leverage their [AI docs](https://ai.google.dev/models)

Reply:
> I understand your concern. In langchain it is seen as a "[provider](https://python.langchain.com/docs/integrations/platforms/)", and each provider is initialized by [importing a specific package](https://python.langchain.com/docs/integrations/platforms/openai), similar [approach for Llamaindex](https://docs.llamaindex.ai/en/stable/getting_started/starter_example_local/). Do you think these could be leveraged?

Yes, we should leverage what we have.
My concern was specifically about azure_openai. If openai SDK is instrumented natively or with an instrumentation library, the only thing instrumentation would know is what this client library is for openai. 

I.e. every vendor that exposes openai-compatible APIs can be behind that endpoint. We can't be more specific that openai in the case.

Reply:
I like the suggestion of @iag of following LlamaIndex and Langchain's footsteps, but let's defer this to a future PR so we can land this first.